### PR TITLE
Adjust Makefile for additional target lavad-protocol

### PR DESCRIPTION
**Pull request by orenl-lava**
User must explicitly choose the target binaries, via the target(s) or using env-var LAVA_BINARY:

{build,install}-lavad: build/install lavad
{build,install}-lava-protocol: build/intsall lava-protocol
{build,install}-all: build/install all binaries
{build,install}: build/install $LAVA_BINARY (in env; error if unset)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -288,35 +333,16 @@ lint:
...
-containerProtoFmt=$(PROJECT_NAME)-proto-fmt-$(protoVer)
+containerProtoGen=lava-proto-gen-$(protoVer)
+containerProtoFmt=lava-proto-fmt-$(protoVer)
 
-proto-gen:
-	@echo "Generating Protobuf files"
-	if docker ps -a --format '{{.Names}}' | grep -Eq "^${containerProtoGen}$$"; then docker start -a $(containerProtoGen); \
-	else docker run --name $(containerProtoGen) -v $(CURDIR):/workspace --workdir /workspace $(protoImageName) \
-		sh ./scripts/protocgen.sh; fi
+proto-gen: $(BUILDDIR)/proto-gen
```
Comment (orenl-lava) [inline Makefile:342]: something got messed up in your merge here - I will fix.

File: lava-protocol/main.go
Comment (orenl-lava) [inline lava-protocol/main.go:1]: should should be `cmd/lava-protocol/main.go` ... I will fix.

**Review by ranlavanet — DISMISSED** [review]

File: lava-protocol/main.go
Comment (orenl-lava) [reply lava-protocol/main.go:1]: fixed 788bc0c

File: Makefile (hunk 1 above)
Comment (orenl-lava) [reply Makefile:363]: fixed 788bc0c

**Review by ranlavanet — DISMISSED** [review]

**Review by omerlavanet — CHANGES_REQUESTED** [review]
If e2e builds we need to remove the build from the e2e workflow otherwise we build twice

**Review by omerlavanet — APPROVED** [review]

**Pull request merged**